### PR TITLE
Re-Introduction of Problem Matchers

### LINKED ISSUE #40: feat: Problem Matchers
https://github.com/ScottBrenner/cfn-lint-action/commit/4e7c712c4d5e271500b8ad3d714bf09eef6e5a58 previously introduced Problem Matchers to this Action, but this functionality was removed with https://github.com/ScottBrenner/cfn-lint-action/commit/15421eb7a3008bd51ffca8ee8fa51296457aa679#diff-efc0e1791ad6067603d6250b9eca6bfa0159eb4f7ce6f65b5ffd68273a179117.

As discussed in https://github.com/ScottBrenner/cfn-lint-action/pull/33/files#r615433016:

>From https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md
> Problem Matchers are a way to scan the output of actions for a specified regex pattern and surface that information prominently in the UI
>
> This will display linting errors inline on pull requests, see the image in ~https://github.com/ScottBrenner/cfn-lint-action#github-action-for-cloudformation-linter~ https://github.com/ScottBrenner/cfn-lint-action/blob/90f970711aa1b19575dd9522d2340c5bca6df803/screenshot.png for an example.
>
> Might have to defer to original implementor @softprops regarding functionality, but in addition to this `cfn-lint.json` file we'll need something similar to https://github.com/ScottBrenner/cfn-lint-action/commit/4e7c712c4d5e271500b8ad3d714bf09eef6e5a58#diff-6f9d41d046756f0ddc2fcee0626bdb50100d12b88f293734eff742818e03efa2R7 looks like.

### PR DESCRIPTION
**Pull request by NickLiffen**
Closes #40

### CODE REVIEW

**Comment by NickLiffen** [comment]
I intend for #38 to get merged before 👍 Hence why I have updated the release for this to be a `v2.1.0` and #38 to be `v2.0.1` 👍 Just FYI.

**Comment by NickLiffen** [comment]
@ScottBrenner I realised the `v2` release changed the name of this GitHub Action. Now links that people have put to this GitHub Action no longer work, so I have reverted it back to the previous name 👍

**Comment by ScottBrenner** [comment]
Oops, yeah see it under https://github.com/marketplace/actions/setup-aws-cfn-lint now 😄

File: dist/index.js [hunk 1]
```diff
@@ -6,12 +6,15 @@ module.exports =
...
 const core = __nccwpck_require__(186);
+const path = __nccwpck_require__(622);
 
 const setup = __nccwpck_require__(391);
 
 (async () => {
   try {
     await setup();
+    const matchersPath = path.join(__dirname, "..", ".github");
+    core.info(`##[add-matcher]${path.join(matchersPath, "cfn-lint.json")}`);
```
Comment (ScottBrenner) [inline dist/index.js:17]: Cool, do you have an example of this working?

File: action.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-name: "Setup AWS CFN LINT"
-description: "Setup AWS CFN LINT and add it to the PATH"
+name: "cfn-lint-action"
+description: "Setup Amazon Cloud Formation Linter (cfn-lint)"
```
Comment (ScottBrenner) [inline action.yml:2]:
```suggestion
description: "Setup Amazon CloudFormation Linter (cfn-lint)"
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.1.0](https://github.com/ScottBrenner/cfn-lint-action/releases/tag/v2.1.0)
+
+### Added
+
+- Cloud Formation Linter [Problem Matchers](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md).
```
Comment (ScottBrenner) [inline CHANGELOG.md:12]:
```suggestion
- Cloud FormationLinter [Problem Matchers](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md).
```

**Comment by ScottBrenner** [comment]
cc: @softprops 😁

File: dist/index.js (hunk 1 above)
Comment (NickLiffen) [reply dist/index.js:17]: I do indeed 💯 Let me upload that now 👍

Comment (NickLiffen) [reply dist/index.js:17]: in my example I have `push` and `pull_request` enabled, hence why it's twice 👍

**Review by ScottBrenner — APPROVED** [review]
LGTM, thanks!

**Comment by ScottBrenner** [comment]
- https://github.com/ScottBrenner/cfn-lint-action/releases/tag/v2
- https://github.com/ScottBrenner/cfn-lint-action/releases/tag/v2.1.0

**Comment by softprops** [comment]
thanks folks 👏

**Comment by softprops** [comment]
I just installed this action at work and verified the problem matchers. This works great!

**Pull request merged**